Lu/extend orange component

luquantan opened the pull request:
For Jira bug 136

luquantan committed:
Move selecDeviceStepView contentMode update animation to resetSelectDeviceView method

luquantan committed:
Added TGKeyboardInfo

luquantan committed:
Remove animationCurve from keyboard info

luquantan committed:
added keyboard notification to TGSelectDeviceStepView

luquantan committed:
Animate selectDeviceStepView with the keyboard

luquantan committed:
Reset content of SelectDeviceStepView in resetSelectDeviceView

anbitasiregar commented on ThreadGroup/Controllers/TGMainViewController.m:
```diff
@@ -199,9 +203,13 @@ - (void)configureMainViewForViewState:(TGMainViewState)viewState {
-            [self resetSelectDeviceView];
-            [self setPopupNotificationForState:viewState animated:YES];
-            break;
+            if ([self isTogglingBetweenScanningAndPassphraseViewStates]) {
```
I was also having the problem where the view was resetting when you were going from no wifi to wifi states, so I put up a quick fix for that in TGRootVC, which can now be deleted.

```
- (void)configureUIForReachableState {
    if (![self.childNavigationController.viewControllers containsObject:self.mainViewController]) {
        [self.mainViewController setViewState:TGMainViewStateLookingForRouters];
        [self.childNavigationController pushViewController:self.mainViewController animated:YES];
    } else {
        if (!(self.mainViewController.viewState == TGMainViewStateConnectDevicePassphrase)) {
            [self.mainViewController setViewState:self.mainViewController.viewState];
        }
        [self.childNavigationController popToViewController:self.mainViewController animated:YES];
    }
}
```

can now be

```
- (void)configureUIForReachableState {
    if (![self.childNavigationController.viewControllers containsObject:self.mainViewController]) {
        [self.mainViewController setViewState:TGMainViewStateLookingForRouters];
        [self.childNavigationController pushViewController:self.mainViewController animated:YES];
    } else {
        [self.mainViewController setViewState:self.mainViewController.viewState];
        [self.childNavigationController popToViewController:self.mainViewController animated:YES];
    }
}
```

luquantan replied:
We don't need to worry about this anymore. We should always return the user to TGMainViewStateLookingForRouter when we configureUIForReachableState

luquantan committed:
Edit configureUIForReachableState to always return user to TGMainViewStateLookingForRouters
When we lose connection to WIFI, we also lose connection to the Thread Network.
The user will need to connect to the network again.

ButkiewiczP commented:
One comment. Other than that, looks good. :+1:
